tracer/src/Datadog.Trace/AppSec/Waf/Result.cs

```diff
@@ -17,13 +16,19 @@ internal class Result : IResult
         public Result(DdwafResultStruct returnStruct, DDWAF_RET_CODE returnCode, ulong aggregatedTotalRuntime, ulong aggregatedTotalRuntimeWithBindings)
         {
             this.returnCode = returnCode;
-            Actions = new((int)returnStruct.ActionsSize);
-            ReadActions(returnStruct);
+            Actions = returnStruct.Actions.DecodeArray<string>();
+            var events = returnStruct.Events.DecodeArray<object>();
+            if (events.Count == 0) { Data = string.Empty; }
+            else
+            {
+                // Serialize all the events
+                Data = JsonConvert.SerializeObject(events);
```

Comment:
FI, and not necessarily part of this PR, but this upgrade allows attack concantenations for us, for example on a double attack like: `http://localhost:54587/health/params/appscan_fingerprint?[$ne]=arg`, waf will run twice for query string then later down the mvc line, path parameters, here we would only report the last one on the span. But might be for a next PR to serialize only at the end, and not set the tag each time at `SecurityCoordinator.Reporter.Report l.77` 👍